benchmark/single-source/InsertCharacter.swift

```diff
@@ -0,0 +1,115 @@
+//===--- InsertCharacter.swift ------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+public let InsertCharacter = [
+    BenchmarkInfo(name: "InsertCharacterEndIndex", runFunction: run_InsertCharacterEndIndex, tags: [.validation, .api], setUpFunction: buildWorkload),
+    BenchmarkInfo(name: "InsertCharacterTowardsEndIndex", runFunction: run_InsertCharacterTowardsEndIndex, tags: [.validation, .api], setUpFunction: buildWorkload),
+    BenchmarkInfo(name: "InsertCharacterStartIndex", runFunction: run_InsertCharacterStartIndex, tags: [.validation, .api], setUpFunction: buildWorkload),
+    BenchmarkInfo(name: "InsertCharacterEndIndexNonASCII", runFunction: run_InsertCharacterEndIndexNonASCII, tags: [.validation, .api], setUpFunction: buildWorkload),
+    BenchmarkInfo(name: "InsertCharacterTowardsEndIndexNonASCII", runFunction: run_InsertCharacterTowardsEndIndexNonASCII, tags: [.validation, .api], setUpFunction: buildWorkload),
+    BenchmarkInfo(name: "InsertCharacterStartIndexNonASCII", runFunction: run_InsertCharacterStartIndexNonASCII, tags: [.validation, .api], setUpFunction: buildWorkload)
+]
+
+let str = String(repeating: "A very long ASCII string.", count: 200)
+
+func buildWorkload() {
+    blackHole(str)
+}
+
+@inline(never)
+func run_InsertCharacterEndIndex(_ N: Int) {
+    var workload = str
+    let character: Character = "a"
+    let scale = 3000
+    for _ in 0..<N * scale {
+        workload.insert(character, at: workload.endIndex)
```

Comment:
I would say: `workload.insert(identity(character), at: identity(workload.endIndex))`. That would make this a little more realistic as then the optimizer cannot know that every iteration of the loop is inserting the exact same character into the exact same (relative) position. Maybe even add a `getCharacter` to TestUtils.swift if the cross-module generic adds any real cost.

Similarly for the other benchmarks, and for the `var workload = str` above. What do you think, @aschwaighofer and @eeckstein ?